packages/@aws-cdk/aws-ec2-alpha/README.md

```diff
@@ -195,9 +213,37 @@ const dynamoEndpoint = new ec2.GatewayVpcEndpoint(this, 'DynamoEndpoint', {
   vpc: myVpc,
   subnets: [subnet],
 });
-new vpc_v2.Route(this, 'DynamoDBRoute', {
+new Route(this, 'DynamoDBRoute', {
   routeTable,
   destination: '0.0.0.0/0',
   target: { endpoint: dynamoEndpoint },
 });
 ```
+
+## Adding Egress-Only Internet Gateway to VPC
+
+An egress-only internet gateway is a horizontally scaled, redundant, and highly available VPC component that allows outbound communication over IPv6 from instances in your VPC to the internet, and prevents the internet from initiating an IPv6 connection with your instances. For more information see@ https://docs.aws.amazon.com/vpc/latest/userguide/egress-only-internet-gateway.html
+
+VPCv2 supports adding an egress only internet gateway to VPC with the help of `addEgressOnlyInternetGateway` method as well.
+
+By Default, it sets up a route to all outbound IPv6 Address ranges unless specified to a specific destination by the user. It can only be set up for IPv6 enabled VPCs.
+`Subnets` takes in value of `SubnetFilter` which can be based on a SubnetType in VPCV2. A new route will be added to route tables of all subnets filtered out with this property.
+
+```ts
+
+const myVpc = new VpcV2(this, 'Vpc');
+const routeTable = new RouteTable(this, 'RouteTable', {
+  vpc: myVpc,
+});
+const subnet = new SubnetV2(this, 'Subnet', {  
+  vpc: myVpc,
+  availabilityZone: 'eu-west-2a',
+  ipv4CidrBlock: new IpCidr('10.0.0.0/24'),
+  subnetType: ec2.SubnetType.PRIVATE });
+
+myVpc.addEgressOnlyInternetGateway({
+  subnets: [{SubnetType.PUBLIC}],
```

Comment:
Haven't looked into implementation but this already looks weird to me.

Reply:
This is as per the current implementations we have in place for addendpoint and addgateways and provides a better way for the users to select multiple subnets at once without forming an array list first and then pass it. This seems to be the best option for specifying a subnet as an input.